keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepositoryTests.kt

```diff
@@ -317,5 +335,33 @@ abstract class CombinedRepositoryTests<D : DeliveryConfigRepository, R : Resourc
         expectCatching { resourceRepository.get(firstResource.id) }.isSuccess()
       }
     }
+
+    context("don't allow storing delivery config with a non unique name") {
+      before {
+        subject.upsertDeliveryConfig(deliveryConfig)
+      }
+      test("trying to persist another config with the same name but different application") {
```

Comment:
I think this should be the description of your `context` above. The `test` here is testing that trying to persist the delivery config `"throws a DuplicateDeliveryConfigsException"`.